### ISSUE TITLE
trace SQL is blocked by restricted read only

### ISSUE THREAD

**Issue by ichn-hu** [pingcap/tidb#31175]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

turn on restricted read only and execute

![image](https://user-images.githubusercontent.com/29735669/147731372-6cec2e85-4e1f-4516-aeb8-ac213f487fb8.png)

### 2. What did you expect to see? (Required)

okay to run

### 3. What did you see instead (Required)

running in read-only mode

### 4. What is your TiDB version? (Required)

master

**Comment by yudongusa** [comment]
@XuHuaiyu assign this to you for now.

**Comment by zanmato1984** [comment]
This seems to be related to readonly mode. Handling to sig/sql-infra.

**Comment by bb7133** [comment]
I don't think this is a major bug: restricted read only mode is not a long-running state of TIDB(for example, it is used to gurantee the upstream cluster will not be changed in 'detached replication' sceanrio), and 'trace ...' is a statement that does not appear in 'online workload'.

I will downgrade the severity to 'moderate'. /cc @aytrack @ichn-hu
